usaspending_api/etl/es_etl_helpers.py

```diff
@@ -0,0 +1,465 @@
+        raise SystemExit
+
+    if config['verbose']:
+        printf({'msg': 'CSV data from {} to now'.format(config['starting_date'])})
+
+    to_datetime = datetime.combine(config['starting_date'], datetime.min.time(), tzinfo=pytz.UTC)
+    filtered_csv_list = [
+        x for x in bucket_objects
+        if (
+            x.key.endswith('.csv') and
+            not x.key.startswith('staging') and
+            x.last_modified >= to_datetime
+        )
+    ]
+
+    if config['verbose']:
+        printf({'msg': 'Found {} csv files'.format(len(filtered_csv_list))})
+
+    deleted_ids = {}
+
+    for obj in filtered_csv_list:
+        # Use temporary files to facilitate date moving from csv files on S3 into pands
+        (file, file_path) = tempfile.mkstemp()
+        bucket.download_file(obj.key, file_path)
+
+        # Ingests the CSV into a dataframe. pandas thinks some ids are dates, so disable parsing
+        data = pd.DataFrame.from_csv(file_path, parse_dates=False)
+
+        if 'detached_award_proc_unique' in data:
+            new_ids = ['cont_tx_' + x for x in data['detached_award_proc_unique'].values]
+        elif 'afa_generated_unique' in data:
+            new_ids = ['asst_tx_' + x for x in data['afa_generated_unique'].values]
+        else:
+            printf({'msg': '  [Missing valid col] in {}'.format(obj.key)})
+
+        # Next statements are ugly, but properly handle the temp files
+        os.close(file)
+        os.remove(file_path)
+
+        for uid in new_ids:
+            if uid in deleted_ids:
+                if deleted_ids[uid]['timestamp'] < obj.last_modified:
+                    deleted_ids[uid]['timestamp'] = obj.last_modified
+            else:
+                deleted_ids[uid] = {'timestamp': obj.last_modified}
+
+    if config['verbose']:
+        for k, v in deleted_ids.items():
+            printf({'msg': 'id: {} last modified: {}'.format(k, str(v['timestamp']))})
+
+    printf({'msg': 'Gathering {} deleted transactions took {}s'.format(len(deleted_ids), perf_counter() - start)})
+    return deleted_ids
+
+
+def filter_query(column, values, query_type="match_phrase"):
+    queries = [{query_type: {column: str(i)}} for i in values]
+    return {
+        "query": {
+            "bool": {
+                "should": [
+                    queries
+                ]
+            }
+        }
+    }
+
+
+def delete_query(response):
+    return {
+        "query": {
+            "ids": {
+                "type": "transaction_mapping",
+                "values": [i['_id'] for i in response['hits']['hits']]
+            }
+        }
+    }
+
+
+def chunks(l, n):
+    """Yield successive n-sized chunks from l."""
+    for i in range(0, len(l), n):
+        yield l[i:i + n]
+
+
+def delete_transactions_from_es(client, id_list, job_id, config, index=None, size=50000):
+    '''
+    id_list = [{key:'key1',col:'tranaction_id'},
+               {key:'key2',col:'generated_unique_transaction_id'}],
```

Comment:
Col is currently hardcoded to be `generated_unique_transaction_id`, will it ever be `tranaction_id`

Reply:
`transaction_id` doesn't work for the records of deleted transactions from the broker stored in S3. While the code ((should)) be able to handle any column, the goal is to use `generated_unique_transaction_id` everywhere. Early tests show that if this field is loaded into ES it appears to work